Modules/_asynciomodule.c

```diff
@@ -646,6 +650,16 @@ future_get_result(FutureObj *fut, PyObject **result)
 
     fut->fut_log_tb = 0;
     if (fut->fut_exception != NULL) {
+        if (fut->fut_exception_tb != NULL) {
+            if (PyException_SetTraceback(fut->fut_exception, fut->fut_exception_tb) < 0)
+            {
+                Py_DECREF(fut->fut_exception_tb);
```

Comment:
Note: `PyException_SetTraceback` takes ownership of the object, so we have no decref responsibility on our side. Also, `fut->fut_exception_tb` points to some object that you just decref'ed. Probably not what you want :)

Reply:
The strange thing is that on windows tests are passing but failing on Linux